src/main/java/storage/FileManager.java

```diff
@@ -0,0 +1,244 @@
+        }
+        fileWriter.close();
+    }
+
+    public void checkIfWorkoutFileAlreadyExists() {
+        if (Files.exists(getWorkoutFilePath())) {
+            setWasWorkoutsFileAlreadyMade(true);
+        }
+    }
+
+    public void createWorkoutFile() throws IOException {
+        Files.createFile(getWorkoutFilePath());
+    }
+
+    /**
+     * Reads the exercises from local file and store them into an ArrayList named exercises.
+     *
+     * @throws IOException when the local file is missing.
+     */
+    public void loadExercisesFromFile(ExerciseList exerciseList) throws IOException {
+        Scanner scanner = new Scanner(getExerciseFilePath());
+        while (scanner.hasNextLine()) {
+            var exercise = scanner.nextLine();
+            exerciseList.addExerciseToList(exercise);
+        }
+    }
+
+    public boolean loadWorkoutsFromFile(WorkoutList workoutList) throws IOException {
+        boolean hasNoErrorsDuringLoad = true;
+        Scanner workoutFileReader = new Scanner(getWorkoutFilePath());
+        while (workoutFileReader.hasNext()) {
+            try {
+                String workoutFileDataLine = workoutFileReader.nextLine();
+                String[] parsedWorkoutFileDataLine = parseFileDataLine(workoutFileDataLine);
+                addFileWorkoutToList(workoutList, parsedWorkoutFileDataLine);
+
+            } catch (ArrayIndexOutOfBoundsException e) {
+                System.out.println("File data error: insufficient parameters in workout data.");
+                hasNoErrorsDuringLoad = false;
+            } catch (InvalidExerciseException | InvalidWorkoutException e) {
+                System.out.println("File data error:" + e.getMessage());
+                hasNoErrorsDuringLoad = false;
+            }
+        }
+
+        return hasNoErrorsDuringLoad;
+    }
+
+    private String[] parseFileDataLine(String fileDataLine) {
+        String[] parsedFileDataLine = fileDataLine.split(FILE_DATA_DELIMITER_REGEX);
+        return parsedFileDataLine;
+    }
+
+    private void addFileWorkoutToList(WorkoutList workoutList, String[] workoutFileDataLine)
+            throws ArrayIndexOutOfBoundsException, InvalidExerciseException, InvalidWorkoutException {
+        String workoutName = workoutFileDataLine[0];
+        String workoutReps = workoutFileDataLine[1];
+
+        String userArguments = workoutName + " " + WorkoutCommand.CREATE_ACTION_REPS_KEYWORD + " " + workoutReps;
+        workoutList.createAndAddWorkout(userArguments);
+    }
+
+    public void writeNewWorkoutToFile(Workout newWorkout) throws IOException {
+        String workoutInFileFormat = convertWorkoutToFileDataFormat(newWorkout);
+
+        FileWriter fileWriter = new FileWriter(getWorkoutFilePath().toString(), true);
+        fileWriter.append(workoutInFileFormat);
+        fileWriter.append(System.lineSeparator());
+        fileWriter.close();
+    }
+
+    private String convertWorkoutToFileDataFormat(Workout workout) {
```

Comment:
Might want to rename this method precisely because from what I see is that this method more of like add the workout into a file? so maybe appendAndConvertWorkoutToFileDataFormat()?

Reply:
This method doesn't add the data into the file itself. It converts the `Workout` data into a format that is meant to be stored in the text file. For example, if the `Workout` object is `russian twist` with `1000` reps, the method will return `russian twist | 1000`. It will not do the writing operation. Hence the name `convertWorkout...` into a `...FileDataFormat`.

The actual writing to the file is done in either the `writeNewWorkoutToFile` or `rewriteAllWorkoutsToFile` methods.

Reply:
I see! Thank you for the clarification :)